adminpages/addons.php

```diff
@@ -22,281 +18,394 @@
+										} else {
+											$action_button['label'] = __( 'Active', 'paid-memberships-pro' );
+											$action_button['url'] = '#';
+											$action_button['style'] = 'button disabled';
+										}
+									}
+
+									if ( is_array( $action_button ) ) { ?>
+										<a href="<?php echo esc_attr( $action_button['url'] ); ?>" class="<?php echo esc_attr( $action_button['style'] ); ?>"><?php echo esc_html( $action_button['label'] ); ?></a>
+									<?php } else {
+										echo $action_button;
+									}
+								?>
+							</div> <!-- end action-button -->
+						</div> <!-- end actions -->
+					</div> <!-- end add-on-item -->
+				</div> <!-- end add-on-container -->
 				<?php
-								
-				ob_start();
-				wp_plugin_update_row( $plugin_file, $plugin_data );
-				$row = ob_get_contents();
-				ob_end_clean();
-				
-				echo str_replace('colspan="0"', 'colspan="4"', $row);
-			}
-		?>
-		</tbody>
-	</table>				
+				}
+			?>
+			</div> <!-- end list -->
+		</div> <!-- end pmpro-admin-add-ons-list -->
+		<script>
+			jQuery(document).ready( function($) {
+				$('[data-search]').keyup(function() {
+					var views = $( '.addons-search' ).closest( '.filter-links' );
+					views.find( 'li a' ).removeClass( 'current' );
+					$('.addons-search a').addClass( 'current' );
+					$('.addons-search').show();
+
+					var filter = $(this).data('search');
+					var filter_items = $(`[data-search-${filter}]`);
+					var search_val = $(this).val();
 
+					if ( search_val != '' ) {
+						filter_items.addClass('search-hide');
+						$(`[data-search-${filter}*="${search_val.toLowerCase()}"]`).removeClass('search-hide');
+					} else {
+						filter_items.removeClass('search-hide');
+						jQuery('.addons-search').hide();
+					}
+				});
+
+				$('.filter-links li a' ).click( function(e) {
+					// don't want to jump to #
+					e.preventDefault();
+
+					var views = $( this ).closest( '.filter-links' );
+					var view = $(this).data('search');
+					var view_items = $(`[data-search-${view}]`);
+					var view_val = $(this).data('view');
+
+					// Update the URL hash.
+					$( this ).attr( 'href' ).replace( /#/, '' );
+
+					// Unstyle view links
+					views.find( 'li a' ).removeClass( 'current' );
+					$( this ).addClass( 'current' );
+					views.find('.addons-search').hide();
+
+					// Clear the search input, if full.
+					document.getElementById( 'search-add-ons' ).value = '';
+
+					// update the URL
+					if ( history.pushState ) {
+					    history.pushState( null, null, '#' + view_val );
+					} else {
+					    location.hash = '#' + view_val;
+					}
+
+					if ( view_val != '' ) {
+						view_items.addClass('search-hide');
+						$(`[data-search-${view}*="${view_val.toLowerCase()}"]`).removeClass('search-hide').addClass('search-show');
+					} else {
+						view_items.removeClass('search-hide').addClass('search-show');
+					}
+
+/*					var n = $( '.search-show' ).length;

```

Comment:
Just some stray commented-out data here :)

Reply:
Removed - thank you!